lib/pbench/server/indexing_tarballs.py

```diff
@@ -0,0 +1,492 @@
+                    # different failures; the rest are going to end up in
+                    # `linkerrdest` for later retry.
+                    controller_path = linksrc_dir.parent
+
+                    if tb_res == 0:
+                        idxctx.logger.info(
+                            "{}: {}/{}: success",
+                            idxctx.TS,
+                            controller_path.name,
+                            os.path.basename(tb),
+                        )
+                        # Success
+                        with indexed.open(mode="a") as fp:
+                            print(tb, file=fp)
+                        rename_tb_link(
+                            tb, Path(controller_path, self.linkdest), idxctx.logger
+                        )
+                    elif tb_res == 1:
+                        idxctx.logger.warning(
+                            "{}: index failures encountered on {}", idxctx.TS, tb
+                        )
+                        with erred.open(mode="a") as fp:
+                            print(tb, file=fp)
+                        rename_tb_link(
+                            tb,
+                            Path(controller_path, f"{self.linkerrdest}.1"),
+                            idxctx.logger,
+                        )
+                    elif tb_res in (2, 3):
+                        assert False, (
+                            f"Logic Bomb!  Unexpected tar ball handling "
+                            f"result status {tb_res:d} for tar ball {tb}"
+                        )
+                    elif tb_res >= 4 and tb_res <= 11:
+                        # # Quietly skip these errors
+                        with skipped.open(mode="a") as fp:
+                            print(tb, file=fp)
+                        rename_tb_link(
+                            tb,
+                            Path(controller_path, f"{self.linkerrdest}.{tb_res:d}"),
+                            idxctx.logger,
+                        )
+                    else:
+                        idxctx.logger.error(
+                            "{}: index error {:d} encountered on {}",
+                            idxctx.TS,
+                            tb_res,
+                            tb,
+                        )
+                        with erred.open(mode="a") as fp:
+                            print(tb, file=fp)
+                        rename_tb_link(
+                            tb, Path(controller_path, self.linkerrdest), idxctx.logger
+                        )
+                    idxctx.logger.info(
+                        "Finished{} {} (size {:d})",
+                        "[SIGQUIT]" if sigquit_interrupt[0] else "",
+                        tb,
+                        size,
+                    )
+
+                    if sigquit_interrupt[0]:
+                        break
+            except SigTermException:
```

Comment:
Do we need to handle SigTermException here?

Reply:
yes this is the most external try/except block so to be cautious

Reply:
@ndokos, see the comment on lines 418 and 419. If there is a better way to word that comment to make this requirement clear, let me know.